### DISCUSSION TITLE
Should HTTP client protocol be async by default?

### DISCUSSION THREAD

**Discussion by bsless** [clj-easy/tools.misc#4] [Ideas]
Title says it all.
A more generic signature would be `[request options success fail]`

wdyt?

**Comment by borkdude** [comment]
@bsless We can have both sync and async functions.

I think it's best to focus on the JSON example first, to limit the scope and decide on a good starting point for this library.
One discussion going on at the moment is this one https://github.com/clj-easy/tools.misc/discussions/3#discussioncomment-2494236
